src/main/scala/com/microsoft/partnercatalyst/fortis/spark/transforms/locations/LocationsExtractorFactory.scala

```diff
@@ -15,14 +15,23 @@ class LocationsExtractorFactory(
   protected var lookup: Map[String, Set[Location]] = _
 
   def buildLookup(): this.type = {
-    val map = mutable.Map[String, mutable.Set[Location]]()
-    featureServiceClient.bbox(geofence).foreach(location => {
-      val key = location.name.toLowerCase
-      val value = toLocation(location)
-      map.getOrElseUpdate(key, mutable.Set()).add(value)
+    val map = mutable.Map[String, Location]()
+    featureServiceClient.bbox(geofence).foreach(feature => {
+      val locationName = feature.name.toLowerCase
+      val newLocation = toLocation(feature)
+      val oldLocation = map.get(locationName).orNull
+      if (oldLocation == null) {
+        logDebug(s"Got new location for name $locationName: ${newLocation.wofId}")
+        map(locationName) = newLocation
+      } else if (newLocation < oldLocation) {
+        logDebug(s"Discarding location ${oldLocation.wofId} for name $locationName as we now have more granular location ${newLocation.wofId}")
+        map(locationName) = newLocation
+      } else {
+        logDebug(s"Discarding location ${newLocation.wofId} for name $locationName since we already have more granular location ${oldLocation.wofId}")
```

Comment:
"Ignoring" may be clearer for ignored incoming locations, especially at a quick glance while debugging (differentiating from a more specific location being found in the logs).

Reply:
7703962